x-pack/plugins/security/server/routes/api_keys/query.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SecurityQueryApiKeysRequest } from '@elastic/elasticsearch/lib/api/types';
+
+import { schema } from '@kbn/config-schema';
+
+import type { RouteDefinitionParams } from '..';
+import type { ApiKey } from '../../../common/model';
+import { wrapIntoCustomErrorResponse } from '../../errors';
+import { createLicensedRouteHandler } from '../licensed_route_handler';
+
+/**
+ * Response of Kibana Query API keys endpoint.
+ */
+export interface QueryApiKeyResult {
+  apiKeys: ApiKey[];
+  canManageCrossClusterApiKeys: boolean;
+  canManageApiKeys: boolean;
+  canManageOwnApiKeys: boolean;
+  count: number;
+  total: number;
+}
+
+export function defineQueryApiKeysRoute({
+  router,
+  getAuthenticationService,
+}: RouteDefinitionParams) {
+  router.post(
+    {
+      path: '/internal/security/api_key/_query',
+      validate: {
+        body: schema.object({
+          query: schema.maybe(schema.object({}, { unknowns: 'allow' })),
+          size: schema.maybe(schema.number()),
+          sort: schema.maybe(schema.object({}, { unknowns: 'allow' })),
+          from: schema.maybe(schema.number()),
```

Comment:
__issue:__ Do I understand correctly that with this API we won't be able to page through more than 10k keys due to default value of `index.max_result_window`? Is it intentional or should we rely on `search_after` instead?

We probably also need test steps to verify how well our UI behaves with 10k+ keys. Let me know if you need any help with that.

Reply:
@azasypkin I think you're right. This doesn't work with more than 10k keys - with this error: `Result window is too large, from + size must be less than or equal to: [10000] but was [11100]. See the scroll api for a more efficient way to request large data sets. This limit can be set by changing the [index.max_result_window] index level setting.`. 

I'll work on changing the `from` and `size` parameters.